src/lib.rs

```diff
@@ -188,7 +196,8 @@ impl<T: Eq + Hash + Send + Sync + Default + 'static> Default for ArcIntern<T> {
 /// hash of the pointer with hash of the data itself.
 impl<T: Eq + Hash + Send + Sync> Hash for ArcIntern<T> {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        Arc::as_ptr(&self.arc).hash(state);
+        let inner: &T = self.arc.deref();
```

Comment:
this is the real change - the rest is just rust-fmt noise.

Reply:
Thanks for the contribution, but I am not convinced this is correct.  The idea was to hash the raw pointer, which is sound since the value is interned and hence all identical copies will point to the same memory location.  Your code will hash the actual value, which is functionally correct, but can be much more expensive if the value is large.

Reply:
Ahhhhhhhhhhhhhhhhhhhhhhhhhhhh. I see. I misunderstood the intent of this!

Reply:
In our case, this is fine, as the hash is trivial - but I totally understand what's going on here now. Feel free to close this PR!

Reply:
Happy to brainstorm other solutions, although at the rate Rust is evolving I find it hard to stay on old versions for long.

I wonder: if we add a feature gate to switch between the two different implementations of `hash`, with the `as_ptr` implementation being the default, will this solve the problem for old compiler versions?